lib/ruboty/adapters/slack_socket_mode.rb

```diff
@@ -85,45 +86,58 @@ def init
       end
 
       def bind
-        realtime.on_text do |data|
-          method_name = "on_#{data['type']}".to_sym
-          send(method_name, data) if respond_to?(method_name, true)
+        socket.on_text do |data|
+          # We could not find whole event types in official document.
+          # This cases will cover all event types found in official sample data.
+          # ref: https://api.slack.com/apis/connections/socket-implement#events
+          case data['type']
+          when 'hello'
+            Ruboty.logger.info("#{self.class.name}: Socket Mode connection is established.")
+          when 'disconnect'
+            # Nothing to do
+            # auto reconnecting works if SLACK_AUTO_RECONNECT configured.
+          when 'events_api'
+            event = data['payload']['event']
+            method_name = "on_#{event['type']}".to_sym
+            send(method_name, event) if respond_to?(method_name, true)
```

Comment:
```suggestion
            event_type = data.dig("payload", "event", "type")
            method_name = :"on_#{event_type}"
            send(method_name, event) if respond_to?(method_name, true)
```

`#dig` だと、ネストしたハッシュのどこかでキーが見つからなかったとき `nil` を返してくれるのでハッシュの構造が変わったときに想定外のわかりづらいエラーにならないです。

`respond_to?` でチェックした結果、対応していないイベントだったらエラーにするなりログを出すなりしておかないとまずい気がします。

Reply:
`dig` とても良いですね・・！
今後もネストしたハッシュを扱うことが増えると想像していますので、多用していこうと思います！
（あと、 `:"文字列"` でシンボルになるのも初めて知りました・・）